Issue with test_adaptative.py

## Issue description

### Current behavior
I have noticed when running tests locally, `test_adaptative.py` always fails, while the tests on GitHub are running. @gisellemartel I believe you had this issue as well?

When the tests are run in `GitHub`, they seem to be running fine. I think the issue might be to do with `hdf5` files and read/write issues. 


### Steps to reproduce
```
pytest testing/unit_tests/test_adaptative.py
```

<details>
<summary> An example summary message </summary>

```
======================================= FAILURES =======================================
______________________________________ test_hdf5 _______________________________________

    def test_hdf5():
        print('[INFO]: Starting test ... \n')
        train_lst = ['sub-unf01']
    
        training_transform_dict = {
            "Resample":
                {
                    "wspace": 0.75,
                    "hspace": 0.75
                },
            "CenterCrop":
                {
                    "size": [48, 48]
                },
            "NumpyToTensor": {}
        }
        transform_lst, _ = imed_transforms.prepare_transforms(training_transform_dict)
    
        roi_params = {"suffix": "_seg-manual", "slice_filter_roi": None}
    
        bids_to_hdf5 = imed_adaptative.BIDStoHDF5(PATH_BIDS,
                                                    subject_lst=train_lst,
                                                    path_hdf5='testing_data/mytestfile.hdf5',
                                                    target_suffix=["_lesion-manual"],
                                                    roi_params=roi_params,
                                                    contrast_lst=['T1w', 'T2w', 'T2star'],
                                                    metadata_choice="contrast",
                                                    transform=transform_lst,
                                                    contrast_balance={},
                                                    slice_axis=2,
                                                    slice_filter_fn=imed_loader_utils.SliceFilter(filter_empty_input=True,
                                                                                        filter_empty_mask=True))
    
        # Checking architecture
        def print_attrs(name, obj):
            print("\nName of the object: {}".format(name))
            print("Type: {}".format(type(obj)))
            print("Including the following attributes:")
            for key, val in obj.attrs.items():
                print("    %s: %s" % (key, val))
    
        print('\n[INFO]: HDF5 architecture:')
        with h5py.File(bids_to_hdf5.path_hdf5, "a") as hdf5_file:
            hdf5_file.visititems(print_attrs)
            print('\n[INFO]: HDF5 file successfully generated.')
            print('[INFO]: Generating dataframe ...\n')
    
            df = imed_adaptative.Dataframe(hdf5_file=hdf5_file,
                                           contrasts=['T1w', 'T2w', 'T2star'],
                                           path='testing_data/hdf5.csv',
                                           target_suffix=['T1w', 'T2w', 'T2star'],
                                           roi_suffix=['T1w', 'T2w', 'T2star'],
                                           dim=2,
                                           filter_slices=True)
    
            print(df.df)
    
            print('\n[INFO]: Dataframe successfully generated. ')
            print('[INFO]: Creating dataset ...\n')
    
            model_params = {
                    "name": "HeMISUnet",
                    "dropout_rate": 0.3,
                    "bn_momentum": 0.9,
                    "depth": 2,
                    "in_channel": 1,
                    "out_channel": 1,
                    "missing_probability": 0.00001,
                    "missing_probability_growth": 0.9,
                    "contrasts": ["T1w", "T2w"],
                    "ram": False,
                    "path_hdf5": 'testing_data/mytestfile.hdf5',
                    "csv_path": 'testing_data/hdf5.csv',
                    "target_lst": ["T2w"],
                    "roi_lst": ["T2w"]
                }
            contrast_params = {
                "contrast_lst": ['T1w', 'T2w', 'T2star'],
                "balance": {}
            }
    
            dataset = imed_adaptative.HDF5Dataset(root_dir=PATH_BIDS,
                                                  subject_lst=train_lst,
                                                  target_suffix="_lesion-manual",
                                                  slice_axis=2,
                                                  model_params=model_params,
                                                  contrast_params=contrast_params,
                                                  transform=transform_lst,
                                                  metadata_choice=False,
                                                  dim=2,
                                                  slice_filter_fn=imed_loader_utils.SliceFilter(filter_empty_input=True,
                                                                                         filter_empty_mask=True),
                                                  roi_params=roi_params)
    
            dataset.load_into_ram(['T1w', 'T2w', 'T2star'])
            print("Dataset RAM status:")
            print(dataset.status)
            print("In memory Dataframe:")
            print(dataset.dataframe)
            print('\n[INFO]: Test passed successfully. ')
    
            print("\n[INFO]: Starting loader test ...")
    
            device = torch.device("cuda:" + str(GPU_NUMBER) if torch.cuda.is_available() else "cpu")
            cuda_available = torch.cuda.is_available()
            if cuda_available:
                torch.cuda.set_device(device)
                print("Using GPU number {}".format(device))
    
            train_loader = DataLoader(dataset, batch_size=BATCH_SIZE,
                                      shuffle=False, pin_memory=True,
                                      collate_fn=imed_loader_utils.imed_collate,
                                      num_workers=1)
    
>           for i, batch in enumerate(train_loader):

testing/unit_tests/test_adaptative.py:135: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
env/lib/python3.8/site-packages/torch/utils/data/dataloader.py:345: in __next__
    data = self._next_data()
env/lib/python3.8/site-packages/torch/utils/data/dataloader.py:856: in _next_data
    return self._process_data(data)
env/lib/python3.8/site-packages/torch/utils/data/dataloader.py:881: in _process_data
    data.reraise()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <torch._utils.ExceptionWrapper object at 0x17cefba30>

    def reraise(self):
        r"""Reraises the wrapped exception in the current thread"""
        # Format a message such as: "Caught ValueError in DataLoader worker
        # process 2. Original Traceback:", followed by the traceback.
        msg = "Caught {} {}.\nOriginal {}".format(
            self.exc_type.__name__, self.where, self.exc_msg)
        if self.exc_type == KeyError:
            # KeyError calls repr() on its argument (usually a dict key). This
            # makes stack traces unreadable. It will not be changed in Python
            # (https://bugs.python.org/issue2651), so we work around it.
            msg = KeyErrorMessage(msg)
>       raise self.exc_type(msg)
E       OSError: Caught OSError in DataLoader worker process 0.
E       Original Traceback (most recent call last):
E         File "/Users/work/Coding/NeuroPoly/ivadomed/env/lib/python3.8/site-packages/torch/utils/data/_utils/worker.py", line 178, in _worker_loop
E           data = fetcher.fetch(index)
E         File "/Users/work/Coding/NeuroPoly/ivadomed/env/lib/python3.8/site-packages/torch/utils/data/_utils/fetch.py", line 44, in fetch
E           data = [self.dataset[idx] for idx in possibly_batched_index]
E         File "/Users/work/Coding/NeuroPoly/ivadomed/env/lib/python3.8/site-packages/torch/utils/data/_utils/fetch.py", line 44, in <listcomp>
E           data = [self.dataset[idx] for idx in possibly_batched_index]
E         File "/Users/work/Coding/NeuroPoly/ivadomed/ivadomed/loader/adaptative.py", line 601, in __getitem__
E           with h5py.File(self.path_hdf5, "r") as f:
E         File "/Users/work/Coding/NeuroPoly/ivadomed/env/lib/python3.8/site-packages/h5py/_hl/files.py", line 406, in __init__
E           fid = make_fid(name, mode, userblock_size,
E         File "/Users/work/Coding/NeuroPoly/ivadomed/env/lib/python3.8/site-packages/h5py/_hl/files.py", line 173, in make_fid
E           fid = h5f.open(name, flags, fapl=fapl)
E         File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
E         File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
E         File "h5py/h5f.pyx", line 88, in h5py.h5f.open
E       OSError: Unable to open file (unable to lock file, errno = 35, error message = 'Resource temporarily unavailable')

env/lib/python3.8/site-packages/torch/_utils.py:395: OSError
```
</details>


## Environment

### System description
MacOSX (Big Sur)
Radeon Pro 560X 4 GB
16 GB 2400 MHz DDR4

### Installed packages
Using `Python 3.8`

<details open>
<summary> Output of <code>pip freeze</code> </summary>

```
absl-py==0.11.0
alabaster==0.7.12
anyio==2.0.2
appnope==0.1.2
argon2-cffi==20.1.0
async-generator==1.10
attrs==20.3.0
Babel==2.9.0
backcall==0.2.0
bids-neuropoly==0.2
bids-validator==1.5.8
bleach==3.2.1
cachetools==4.2.0
certifi==2020.12.5
cffi==1.14.4
chardet==4.0.0
click==7.1.2
csv-diff==1.0
cycler==0.10.0
decorator==4.4.2
defusedxml==0.6.0
dictdiffer==0.8.1
docopt==0.6.2
docutils==0.16
entrypoints==0.3
flake8==3.8.4
future==0.18.2
google-auth==1.24.0
google-auth-oauthlib==0.4.2
grpcio==1.34.0
h5py==2.10.0
idna==2.10
imageio==2.9.0
imagesize==1.2.0
iniconfig==1.1.1
ipykernel==5.4.2
ipython==7.19.0
ipython-genutils==0.2.0
ipywidgets==7.6.3
-e git+https://github.com/ivadomed/ivadomed.git@546c023f23f181da6cc5385e6b2074be61d74245#egg=ivadomed
jedi==0.17.0
Jinja2==2.11.2
joblib==1.0.0
json5==0.9.5
jsonschema==3.2.0
jupyter==1.0.0
jupyter-client==6.1.10
jupyter-console==6.2.0
jupyter-core==4.7.0
jupyter-server==1.2.0
jupyterlab==3.0.1
jupyterlab-pygments==0.1.2
jupyterlab-server==2.1.0
jupyterlab-widgets==1.0.0
kiwisolver==1.3.1
Markdown==3.3.3
MarkupSafe==1.1.1
matplotlib==3.3.3
mccabe==0.6.1
mistune==0.8.4
nbclassic==0.2.5
nbclient==0.5.1
nbconvert==6.0.7
nbformat==5.0.8
nest-asyncio==1.4.3
networkx==2.5
nibabel==3.2.1
notebook==6.1.6
num2words==0.5.10
numpy==1.19.5
oauthlib==3.1.0
onnxruntime==1.4.0
packaging==20.8
pandas==1.2.0
pandocfilters==1.4.3
parso==0.8.1
patsy==0.5.1
pexpect==4.8.0
pickleshare==0.7.5
Pillow==8.1.0
pluggy==0.13.1
prometheus-client==0.9.0
prompt-toolkit==3.0.10
protobuf==3.14.0
ptyprocess==0.7.0
py==1.10.0
pyasn1==0.4.8
pyasn1-modules==0.2.8
pybids==0.12.4
pycodestyle==2.6.0
pycparser==2.20
pyflakes==2.2.0
Pygments==2.7.3
pyparsing==2.4.7
pyrsistent==0.17.3
pytest==6.2.1
python-dateutil==2.8.1
pytz==2020.5
PyWavelets==1.1.1
pyzmq==20.0.0
qtconsole==5.0.1
QtPy==1.9.0
requests==2.25.1
requests-oauthlib==1.3.0
rsa==4.6
scikit-image==0.18.1
scikit-learn==0.24.0
scipy==1.6.0
seaborn==0.11.1
Send2Trash==1.5.0
six==1.15.0
sniffio==1.2.0
snowballstemmer==2.0.0
Sphinx==3.4.3
sphinx-rtd-theme==0.5.1
sphinxcontrib-applehelp==1.0.2
sphinxcontrib-devhelp==1.0.2
sphinxcontrib-htmlhelp==1.0.3
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-programoutput==0.16
sphinxcontrib-qthelp==1.0.3
sphinxcontrib-serializinghtml==1.1.4
SQLAlchemy==1.3.22
tensorboard==2.4.0
tensorboard-plugin-wit==1.7.0
terminado==0.9.2
testpath==0.4.4
threadpoolctl==2.1.0
tifffile==2020.12.8
toml==0.10.2
torch==1.5.0
torchvision==0.6.0
tornado==6.1
tqdm==4.55.1
traitlets==5.0.5
urllib3==1.26.2
wcwidth==0.2.5
webencodings==0.5.1
Werkzeug==1.0.1
widgetsnbextension==3.5.1
```

</details>

--- a/docs/source/installation.rst
+++ b/docs/source/installation.rst
@@ -6,7 +6,12 @@ Supported OS
 
 Currently, we only support ``MacOS`` and ``Linux`` operating systems. ``Windows``
 users have the possibility to install and use ``ivadomed`` via
-`Windows Subsystem for Linux (WSL) <https://docs.microsoft.com/en-us/windows/wsl/>`_.
+`Windows Subsystem for Linux (WSL) <https://docs.microsoft.com/en-us/windows/wsl/>`_. For MacOs users, we strongly recommend to follow the bellow steps before the installation.
+::
+
+    vim ~/.bashrc(.zshrc if you are using zsh shell)
+    (write in the .bashrc file => export HDF5_USE_FILE_LOCKING='FALSE')
+    source .bashrc
 
 Python
 ------

--- a/docs/source/installation.rst
+++ b/docs/source/installation.rst
@@ -11,7 +11,7 @@ users have the possibility to install and use ``ivadomed`` via
 
     vim ~/.bashrc(.zshrc if you are using zsh shell)
     (write in the .bashrc file => export HDF5_USE_FILE_LOCKING='FALSE')
-    source .bashrc
+    source ~/.bashrc
 
 Python
 ------

--- a/docs/source/installation.rst
+++ b/docs/source/installation.rst
@@ -6,12 +6,26 @@ Supported OS
 
 Currently, we only support ``MacOS`` and ``Linux`` operating systems. ``Windows``
 users have the possibility to install and use ``ivadomed`` via
-`Windows Subsystem for Linux (WSL) <https://docs.microsoft.com/en-us/windows/wsl/>`_. For MacOs users, we strongly recommend to follow the bellow steps before the installation.
+`Windows Subsystem for Linux (WSL) <https://docs.microsoft.com/en-us/windows/wsl/>`_.For MacOs users, we strongly recommend to follow the bellow steps before the installation.
+
+Open your bash/zsh script file with editor on your computer.
+
+    If you are using bash shell
+    ::
+
+        vim ~/.bashrc
+
+    If you are using zsh shell
+    ::
+        vim ~/.zshrc
+
+Write in your .bashrc/.zshrc file with following line.
+
 ::
 
-    vim ~/.bashrc(.zshrc if you are using zsh shell)
-    (write in the .bashrc file => export HDF5_USE_FILE_LOCKING='FALSE')
-    source ~/.bashrc
+    export HDF5_USE_FILE_LOCKING='FALSE'
+
+Save this change and restart your terminal to apply the change.
 
 Python
 ------

